Kernel/Arch/mcontext.h

```diff
@@ -10,4 +10,6 @@
 #    include <Kernel/Arch/x86_64/mcontext.h>
 #elif defined(__aarch64__)
 #    include <Kernel/Arch/aarch64/mcontext.h>
+#elif defined(__riscv) && __riscv_xlen == 64
+#    include <Kernel/Arch/riscv64/mcontext.h>
```

Comment:
You make sure that this is only included on RV64, but then you use the pointer-sized integer type in the implementation anyways. This doesn't make sense IMO, it's fine to use uint64_t in RV64's mcontext (coincidentally making it identical to aarch64).